File: robin/python/programmers/level2/study3/get_min_sum.py
def solution(A, B):
    answer = 0

    A = sorted(A)
    B = sorted(B, reverse=True)

    for a, b in zip(A, B):
        answer += a * b

    return answer


if __name__ == '__main__':
    A = [1, 4, 2]
    B = [5, 4, 4]
    print(solution(A, B))

File: robin/java/src/programmers/level2/study3/GetMinSum.java
package programmers.level2.study3;

import java.util.Arrays;

public class GetMinSum {

    public static long solution(int[] A, int[] B) {
        long answer = 0;

        A = Arrays.stream(A).sorted().toArray();
        B = Arrays.stream(B).sorted().toArray();


        for (int i = 0; i < A.length; i++) {
            answer += A[i] * B[(B.length - 1) - i];
        }

        return answer;
    }

    public static void main(String[] args) {
        int[] A = {1, 4, 2};
        int[] B = {5, 4, 4};
        System.out.println(solution(A, B));
    }
}
